javascript/javascript3/week2/promise-visual/main.js

```diff
@@ -0,0 +1,33 @@
+const redCircle = document.querySelector("ul.marks li:nth-child(1)");
+const blueCircle = document.querySelector("ul.marks li:nth-child(2)");
+const greenCircle = document.querySelector("ul.marks li:nth-child(3)");
+
+// translateOneByOne function
+
+function translateOneByOne() {
+  moveElement(redCircle, { x: 20, y: 300 })
+    .then(() => {
+      console.log("Redcircle has been moved");
+    })
+    .then(() => moveElement(blueCircle, { x: 400, y: 300 }))
+    .then(() => {
+      console.log("Bluecircle has been moved");
+    })
+    .then(() => moveElement(greenCircle, { x: 400, y: 20 }))
+    .then(() => {
+      console.log("Greencircle has been moved");
+    });
+}
+
+// translateAllAtOnce function
+
+function translateAllAtOnce() {
+  Promise.all([
+    moveElement(redCircle, { x: 20, y: 300 }),
+    moveElement(blueCircle, { x: 400, y: 300 }),
+    moveElement(greenCircle, { x: 400, y: 20 }),
+  ]).then(() => console.log("All circles has been moved"));
```

Comment:
`.then` can be moved to the user of the promise:
`translateAllAtOnce().then(() => console.log("All circles has been moved"));`

Reply:
function translateAllAtOnce() {
  return  Promise.all([
        moveElement(redCircle, {
            x: 20,
            y: 300
        }),
        moveElement(blueCircle, {
                x: 400,
                y: 300
            },
            moveElement(greenCircle, {
                x: 400,
                y: 20
            })),
    ])
};
translateAllAtOnce()
    .then(() => {
        console.log('All circles have been moved');
    })